core/src/networked_server.cpp

```diff
@@ -0,0 +1,176 @@
+#include <knoting/engine.h>
+#include <knoting/networked_server.h>
+
+namespace knot {
+NetworkedServer::~NetworkedServer() {}
+
+NetworkedServer::NetworkedServer(Engine& engine) : m_engine(engine), m_server(nullptr) {}
+void NetworkedServer::on_awake() {
+    uint8_t privateKey[KeyBytes] = {0};
+    m_server = std::make_shared<Server>(GetDefaultAllocator(), privateKey, Address(SERVER_ADDRESS, SERVER_PORT),
+                                        m_config, gameAdapter, get_time());
+    m_server->Start(MAX_CLIENTS);
+    log::debug("STARTED SERVER...");
+    seq.fill(0);
+    cliSeq.fill(0);
+    cliAck.fill(0);
+}
+void NetworkedServer::on_update(double m_delta_time) {
+    if (!m_server->IsRunning()) {
+        log::debug("server not running!");
+        return;
+    }
+
+    m_server->AdvanceTime(get_time());
+    m_server->ReceivePackets();
+
+    m_tickTime += m_delta_time;
+    if (m_server->GetNumConnectedClients() > 0) {
+        handle_recieved_packets(m_delta_time);
+        send_message();
+    } else {
+        seq.fill(0);
+    }
+    if (m_tickTime >= TICK) {
+        m_tickTime -= TICK;
+    }
+    m_server->SendPackets();
+}
+void NetworkedServer::on_destroy() {
+    m_server->DisconnectAllClients();
+    m_server->Stop();
+    m_server = nullptr;
+}
+double NetworkedServer::get_time() {
+    auto engOpt = Engine::get_active_engine();
+    if (engOpt) {
+        return engOpt.value().get().get_current_time();
+    }
+}
+bool NetworkedServer::send_message() {
+    if (m_tickTime >= TICK) {
+        for (int i = 0; i < m_server->GetNumConnectedClients(); ++i) {
+            if (m_server->CanSendMessage(i, 1)) {
+                Message* mess = generateMessage(i);
+                seq[i]++;
+                m_server->SendMessage(i, 1, mess);
+            }
+        }
+    }
+    return true;
+}
+bool NetworkedServer::handle_recieved_packets(double m_delta_time) {
+    if (!m_server->IsRunning()) {
+        return false;
+    }
+    if (!(m_server)) {
+        return false;
+    }
+    Message* mess = nullptr;
+    for (int i = 0; i < m_server->GetNumConnectedClients(); ++i) {
+        while ((mess = m_server->ReceiveMessage(i, 1))) {
```

Comment:
A do while would be more readable

Reply:
Same as in the client with the releaseMessage this becomes very difficult to do